python/HeaderService/hscalc.py

```diff
@@ -0,0 +1,105 @@
+# This file is part of HeaderService
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This module contains the function used by the HeaderService to computed the
+metadata not provided as SAL/DDS Telemetry/Event
+
+"""
+
+from astropy.time import Time
+import astropy.units as u
+from astropy.coordinates import AltAz, ICRS, EarthLocation
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_date_utc(timeStamp=None):
+
+    """
+    A simple function to the get an UTC astropy.datetime.Time object
+
+    Parameters
+    ----------
+
+    timeStamp : float
+        Optional timestamp (in seconds) passed as an argument. If None
+        is received, then the function will get the time now()
+
+    Returns
+    -------
+
+    t: type of astropy.datetime.Time object
+       The astropy.datetime.Time object with the timeStamp requested in UTC.
+
+    """
+
+    if timeStamp is None:
+        t = Time.now()
+    else:
+        t = Time(timeStamp, format="unix")
+    return t
+
+
+def get_radec_from_altaz(alt, az, obstime, lat=-30.244639, lon=-70.749417, height=2663.0):
```

Comment:
Where are you storing the real values for lat/long ? They should be stored elsewhere and always passed rather than rely on defaults.

Reply:
They should definitely be guaranteed to match the OBSGEO values in the header. We can't afford to get into odd inconsistencies where the header has one location and the derived values use different values.  In an ideal world the pointing task would be using the same telescope location as everyone else so the place the pointing task gets the location should be the same place the header service gets it from (and if we ever change that location it should be part of configuration control and changed in a single place).

Reply:
Now it is passing the values defined on the primary header template.